usb/ehci/ehci.c

```diff
@@ -34,32 +34,42 @@
 
 #include "ehci.h"
 
+#ifdef EHCI_IMX
 #define EHCI_PERIODIC_SIZE 128
+#else
+#define EHCI_PERIODIC_SIZE 1024
+#endif
 
 
 static inline void ehci_memDmb(void)
 {
-	asm volatile ("dmb" ::: "memory");
+#ifdef EHCI_IMX
+	asm volatile("dmb" ::: "memory");
+#else
+	__sync_synchronize();
+#endif
```

Comment:
https://godbolt.org/z/1q4s7Gsbe
```suggestion
	__sync_synchronize();
```

Reply:
GCC won't reduce `__sync_synchronize()` to `dmb` prior to 9.3.1, see e.g. 9.2.1 https://godbolt.org/z/orPnWevvn

Reply:
For future reference: we are in the process of migrating to GCC 14, so @agkaminski's suggestion can be applied once we move completely and probably in a new PR (separating platform-dependent parts in per-arch files).